tests/infrastructure/test_infrastructure_module.py

```diff
@@ -0,0 +1,14 @@
+from opyoid import Injector
+
+from erica.infrastructure.infrastructure_module import InfrastructureModule
+from erica.infrastructure.sqlalchemy.database import DatabaseSessionProvider
+from erica.infrastructure.sqlalchemy.repositories.erica_request_repository import EricaRequestRepository
+
+
+class TestInfrastructureModule:
+
+    def test_if_used_to_inject_erica_req_repository_then_db_connection_correct_injected(self):
+        injector = Injector([InfrastructureModule()])
+        erica_request_repository = injector.inject(EricaRequestRepository)
+
+        assert erica_request_repository.db_connection.bind == DatabaseSessionProvider().get().bind
```

Comment:
Yes, this would be tested via integration, too. However, I think there are two reasons why I would test that here:

- It makes it easier to debug. If the integration test fails, it takes time to figure out why something does not work. These tests give us the confidence that using the InfrastructureModule everything is correctly injected and that changes to some parts of the infrastructure does not lead to any problems with this. 
- Tests should always also be a way to document how we intend that our code is used. The usage of the opyoid library is not really straightforward. It took me quite some time (and these tests tbh) to figure out how it works. I would hope that this makes that more easy